go/pkg/client/cas.go

```diff
@@ -135,7 +142,26 @@ func (c *Client) WriteProto(ctx context.Context, msg proto.Message) (digest.Dige
 func (c *Client) WriteBlob(ctx context.Context, blob []byte) (digest.Digest, error) {
 	ch := chunker.NewFromBlob(blob, int(c.ChunkMaxSize))
 	dg := ch.Digest()
-	return dg, c.WriteChunked(ctx, c.ResourceNameWrite(dg.Hash, dg.Size), ch)
+
```

Comment:
That is not enough to change if we want all writes to be compressed. Things don't usually go through this intermediate-layer function. Most use-cases use UploadIfMissing. I'd suggest creating a Client wrapper function for creating a Chunker from file (which initializes the correct type of chunker based on client parameters) and use it everywhere that Chunkers are created.